app/Http/Controllers/AdminController.php

```diff
@@ -3,24 +3,123 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use App\Models\Scout;
+use App\Models\Preference;
+use App\Models\Program;
+use App\Models\Session;
+use Illuminate\Support\Facades\Log;
 
 class AdminController extends Controller
 {
 
     public function import_data() {
-        
+        $inputFileName = '/home/isaac/importData.xlsx'; // TODO
+        $spreadsheet = IOFactory::load($inputFileName);
+        $sheetData = $spreadsheet->getActiveSheet()->toArray(null, true, true, true);
+        $sheetData = array_slice($sheetData, 3);
+        foreach($sheetData as $row){
+            if ($row['A'] == null) { // Name should never be empty, so this is an empty row. Skip it.
+                continue;
+            }
+            echo "creating... " . implode(", ", $row);
+            $scout = Scout::where('first_name', $row['B'])->where('last_name', $row['C'])->where('unit', $row['J'])->firstOr(function() use ($row) {
+                $scout = new Scout;
+                $scout->first_name = $row['B'];
+                $scout->last_name = $row['C'];
+                if ($row['D'] != null)
+                    $scout->rank = $row['D'];
+                if ($row['F'] != null)
+                    $scout->age = $row['F'];
+                if ($row['G'] != null)
+                    $scout->grade = $row['G'];
+                if ($row['H'] != null)
+                    $scout->years_at_camp = $row['H'];
+                $scout->unit = $row['J'];
+                $scout->site = $row['K'];
+                $scout->save();
+                return $scout;
+            });
+            
+            //create preference
+            preg_match('/(.*) \((\d*)(?:st|nd|rd|th) Pref\)/', $row['A'],  $regex_result);
+            $program = Program::where('name', $regex_result[1])->first();
```

Comment:
Not sure how good we're trusting our data to be, here, but it might be worth catching the case where somebody puts improperly formatted data in and the regex doesn't match (say `1st pref` instead of `1st Pref`)